fleetspeak/src/server/db/store.go

```diff
@@ -71,6 +73,20 @@ type ClientData struct {
 	Blacklisted bool
 }
 
+// Clone returns a deep copy.
+func (d *ClientData) Clone() *ClientData {
```

Comment:
When adding new fields to the ClientData type, we need to remember to update Clone() - that is something that is easy to overlook. It would be nice if there was an automated way of ensuring all members of the struct get copied over.

Reply:
I agree, but haven't thought of a good way to make this happen, beyond putting it right next to the struct definition. Pulling as is for now, but happy to brainstorm ways to make this better in a follow up PR.